sn_node/src/put_validation.rs

```diff
@@ -30,48 +29,135 @@ use std::collections::{BTreeSet, HashSet};
 use tokio::task::JoinSet;
 
 impl Node {
-    /// Validate and store a record to the RecordStore
+    /// Validate a record and it's payment, and store the record to the RecordStore
     pub(crate) async fn validate_and_store_record(
         &self,
         record: Record,
-        validate_payment: bool,
     ) -> Result<CmdOk, ProtocolError> {
         let record_header = RecordHeader::from_record(&record)?;
 
+        if !record_header.includes_payment() {
+            return Err(ProtocolError::PaymentProofInsufficientAmount {
+                paid: Token::zero(),
+                expected: Token::zero(),
+            });
```

Comment:
Here this error message might be confusing, why don't we create another error type like `RecordMissingPayment`?

Reply:
i've dropped this entirely with the dbcspendwithpayment removal 👍